sympy/core/relational.py

```diff
@@ -434,7 +441,11 @@ def __new__(cls, lhs, rhs=None, **options):
                 if L is False:
                     if lhs == -rhs:  # Eq(oo, -oo)
                         return S.false
-                    return S.true
+                    if lhs.is_extended_real and rhs.is_extended_real:
+                        L, R = [i.is_extended_positive for i in (lhs, rhs)]
+                        if L == R and L != None:
+                            return S.true
```

Comment:
Not a big deal but I guess since these are already known to be infinite this could be written more simply as
```julia
if (lhs.is_extended_positive and rhs.is_extended_positive
        or lhs.is_extended_negative and rhs.is_extended_negative):
    return S.true
```
This is another of those cases I think where I think the extended_real infinities are awkward because we don't have an assumption that directly corresponds to each of them. I think that the old assumptions can be tidied up a lot by having `is_posinf` and `is_neginf` to specifically represent these two values: https://github.com/sympy/sympy/pull/17118#issuecomment-511746093